src/Symfony/Component/HttpClient/ServerSentEvents/EventSource.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Symfony\Component\HttpClient\ServerSentEvents;
+
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+final class EventSource implements EventSourceInterface
+{
+    private $url;
+    private $client;
+    private $reconnectionTime = 5;
+    private $shouldClose = false;
+    private $lastEventId = null;
+    private $response = null;
+
+    const CONNECTING = 0;
+    const OPEN = 1;
+    const CLOSED = 2;
+
+    public $readyState = self::CLOSED;
+
+    public function __construct(string $url, array $defaultOptions = [], HttpClientInterface $client = null)
+    {
+        $this->url = $url;
+        $this->client = $client ?: HttpClient::create();
+    }
+
+    public function getHttpClient(): HttpClientInterface
+    {
+        return $this->client;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getMessages(): \Iterator
```

Comment:
would it make sense to allow monitoring several streams at once? This signature forbids it for now while the client can do it.